skycoin/skywire
# Fix vpn client

Did you run `make format && make check`?
yes

Fixes #783 
Fixes #900

 Changes:	
- Vpn retirer only trys 3 times
- Added errors from server hello to retire whitelist
- Removed conSummarry check in AppStates()
- Added new app state `AppStatusErrored` : ` "status": 2` which gives the error inside `detailed_status`
- Added a transport check before dailing routes adapted from [here](https://github.com/alexadhy/skywire/commit/916f90c73d79a6660cd17a16cb77b42bc645d843)
- Added `RPCErr` to preserve the type of the error we return via RPC (as suggested by @alexadhy) 

How to test this PR:
Test #783 
1. Run visorA with VPN server and passcode
2. Run visorB create a transport to vpn-server and connect to Vpn-server of visor A with the wrong passcode.
3. Visor won't retry after the first try.
4. Check `http://localhost:8000/api/visors/<pk>/summary` and see if `vpn-client` has
```json
{
  "name": "vpn-client",
  "args": [
    "-srv",
    "02d27b250feea1fc2fca518baff5eb30df0d77d43fd2cc323d1a25be6ce3abc7be"
  ],
  "auto_start": false,
  "port": 43,
  "status": 2,
  "detailed_status": "Password didn't match"
}
```

Test #900
1. Run visorB without creating a transport to vpn-server and connect to Vpn-server of visor A.
2. Visor won't retry after the first try.
3. Check `http://localhost:8000/api/visors/<pk>/summary` and see if `vpn-client` has
```json
{
  "name": "vpn-client",
  "args": [
    "-srv",
    "02d27b250feea1fc2fca518baff5eb30df0d77d43fd2cc323d1a25be6ce3abc7be"
  ],
  "auto_start": false,
  "port": 43,
  "status": 2,
  "detailed_status": "transport not found"
}
```
